### PR TITLE
Update docs for moved test locations

### PR DESCRIPTION
**Pull request by hamishwillee**
This updates docs for https://github.com/intel/camera-streaming-daemon/pull/138

### CODE REVIEW

**Comment by hamishwillee** [comment]
@SaliniVenate @lbegani This just updates test docs to reflect that the docs have moved. Does raise some questions:
1. test-camera-parameters
  * This was previously undocumented.
  * My testing shows that if you use the set-get option the round trip succeeds irrespective of what values you put in. However if you do a set followed by a get you end up with gibberish. 

I had assumed that this might be for testing set/get of camera specific parameters but it doesn't allow you to specify a camera and it doesn't care what values you set, so it can't be for that. 

Can you let me know what people are supposed to do with this?

2. The original samples had `./samples/camera-sample-custom` - what has happened with this?

**Comment by lbegani** [comment]
@hamishwillee  Moved and Renamed test sources to make it sound better - https://github.com/intel/camera-streaming-daemon/pull/138 . What would be the best way to keep you in loop about such changes, add you as reviewer?
> test-camera-parameters

`samples/main_sample_camparams_client.cpp → test/test_camera_parameters.cpp`

test-camera-parameters was created just to unit test the camera parameters class and not CSD. The reason I created it was to ensure that the storage and retrieval of different data types (stored as string) from map DS works well. Perhaps not very useful for end users. I usually like to create such unit test code for each new class to ensure the APIs work, encapsulation, resusability etc. 

> The original samples had ./samples/camera-sample-custom - what has happened with this?

Renamed it to test-custom-stream
```
-EXTRA_PROGRAMS = samples/camera-sample-custom
+EXTRA_PROGRAMS = test/test-custom-stream
```

File: en/test/test_apps.md [hunk 1]
```diff
@@ -78,7 +78,35 @@ And if *gstreamer* client *gst-launch-1.0* is installed, the selected video will
 
 <!-- Open issue: https://github.com/intel/camera-streaming-daemon/issues/132 -->
 
-## Demo code for Extending CSD (camera-sample-custom)
+
+## Test Camera Parameters (test-camera-parameters)
```
Comment (lbegani) [inline en/test/test_apps.md]: Description needs to be updated based on comment made

**Comment by hamishwillee** [comment]
> @hamishwillee Moved and Renamed test sources to make it sound better - intel/camera-streaming-daemon#138 . What would be the best way to keep you in loop about such changes, add you as reviewer?

Yes, or even better add "Docs" label and add a note to me @hamishwillee that docs changes expected.

**Comment by hamishwillee** [comment]
>> The original samples had ./samples/camera-sample-custom - what has happened with this?

> Renamed it to test-custom-stream

OK, I have updated the name. Can you please re-review. I'm only guessing at the description. For example, I have said it is example code for the guide showing how to create custom streams, but actually it might just be test code that works with the guide instructions (ie if you followed the guide and then ran the test you would be able to test it). Can you confirm the "exact" purpose.

**Comment by hamishwillee** [comment]
> test-camera-parameters was created just to unit test the camera parameters class and not CSD. The reason I created it was to ensure that the storage and retrieval of different data types (stored as string) from map DS works well. Perhaps not very useful for end users. I usually like to create such unit test code for each new class to ensure the APIs work, encapsulation, resusability etc.

Make perfect sense. I have put under a "Unit tests" heading and added your description. Can you please review.

As an aside though, unit tests make a lot of sense, but it is usual to embed these in a suite based on some test framework, and run them on CI in every PR. 
- Do we do that with this test? 
- Do we have other similar tests?

**Comment by lbegani** [comment]
> Do we do that with this test?

No, we do not. But its a good TODO task.
> Do we have other similar tests?

No. Idea was there, but didnt execute it.

**Comment by lbegani** [comment]
LGTM

**Comment by hamishwillee** [comment]
Thanks @lbegani . I have created a todo for tests: https://github.com/intel/camera-streaming-daemon/issues/145

**Pull request merged**